terraform/modules/daily_snapshot/service/list_snapshots.rb

```diff
@@ -12,23 +12,33 @@
 class Snapshot
   attr_accessor :network, :date, :height, :file_name, :url_s3, :url
 
-  def initialize(network, date, height, file_name, url_s3, url)
-    if network.nil? || date.nil? || height.nil? || file_name.nil? || url_s3.nil? || url.nil?
-      raise ArgumentError, 'Missing argument'
-    end
+  def initialize(params)
+    check_for_nil_params(params)
+    check_network_param(params)
 
-    network = network.downcase
+    @network = params[:network].downcase
+    @date = params[:date].to_date
+    @height = params[:height].to_i
+    @file_name = params[:file_name]
+    @url_s3 = params[:url_s3]
+    @url = params[:url]
+  end
 
-    if network != 'mainnet' && network != 'calibnet'
-      raise ArgumentError, 'Invalid network'
+  def check_for_nil_params(params)
+    unless params[:network].nil? || params[:date].nil? || params[:height].nil? || \
+           params[:file_name].nil? || params[:url_s3].nil? || params[:url].nil?
+      return
     end
 
-    @network = network
-    @date = date.to_date
-    @height = height.to_i
-    @file_name = file_name
-    @url_s3 = url_s3
-    @url = url
+    raise ArgumentError,
+          'Missing argument'
+  end
+
+  def check_network_param(params)
+    return unless params[:network].downcase != 'mainnet' && params[:network].downcase != 'calibnet'
+
+    raise ArgumentError,
+          'Invalid network'
```

Comment:
It'd be useful to print the name of the network that is not valid, for debugging purposes